Error message when the process stops on its own

Some implementations, e.g. SuiteSparse:GraphBLAS exit upon finishing. Therefore, the framework throws an error stating that it couldn't kill the process:

```console
07:38 [WARN ] Terminating process 13674 focibly.
07:38 [WARN ] Executing command "kill -9 13674"
07:38 [ERROR] Failed to kill runner process.
```

This might be confusing, especially for newcomers. It's definitely annoying for regulars as well :-).
Is there a way to turn this off?

--- a/graphalytics-core/src/main/java/science/atlarge/graphalytics/execution/BenchmarkExecutor.java
+++ b/graphalytics-core/src/main/java/science/atlarge/graphalytics/execution/BenchmarkExecutor.java
@@ -491,7 +491,7 @@ public class BenchmarkExecutor {
 			}
 			BenchmarkRunner.terminateRunner(runnerInfo);
 			runnerInfo.setTerminated(true);
-			LOG.info(String.format("The benchmark run is sucessfully terminated."));
+			LOG.info(String.format("The benchmark run is successfully terminated."));
 		} catch (Exception e) {
 			LOG.error("Failed to terminate benchmark run.");
 			throw new GraphalyticsExecutionException("Benchmark is aborted.", e);

--- a/graphalytics-core/src/main/java/science/atlarge/graphalytics/execution/BenchmarkRunner.java
+++ b/graphalytics-core/src/main/java/science/atlarge/graphalytics/execution/BenchmarkRunner.java
@@ -209,37 +209,28 @@ public class BenchmarkRunner {
 			throw new GraphalyticsExecutionException("Failed to find benchmark runner registration. Benchmark aborted.", e);
 		}
 
-		// First attempt to terminate runner process gracefully.
-		LOG.debug("Terminating runner process gracefully.");
-		ProcessUtil.terminateProcess(process);
-
-		boolean terminated = ProcessUtil.isNetworkPortAvailable(port) && !ProcessUtil.isProcessAlive(processId);
-
-		LOG.debug(String.format("Runner process is %s: (process alive=%s, port available=%s)",
-				terminated ? "terminated" : "alive",
-				ProcessUtil.isProcessAlive(processId),
-				ProcessUtil.isNetworkPortAvailable(port)));
-
-		while (!terminated) {
-			LOG.warn("Terminating runner process forcibly.");
+		while (!ProcessUtil.isProcessTerminated(port, processId)) {
+			LOG.debug("Terminating runner process.");
 			try {
 				ProcessUtil.terminateProcess(processId);
 			} catch (Exception e) {
 				LOG.error("Failed to terminated runner process.", e);
 			}
 
-			if(!terminated) {
+			LOG.info("Checking if the runner process has been successfully terminated..");
+			TimeUtil.waitFor(10);
+
+			if (!ProcessUtil.isProcessTerminated(port, processId)) {
 				LOG.error(String.format("Failed to kill runner process."));
-				TimeUtil.waitFor(10);
 			}
-			terminated = ProcessUtil.isNetworkPortAvailable(port) && !ProcessUtil.isProcessAlive(processId);
 
 			LOG.debug(String.format("Runner process is %s: (process alive=%s, port available=%s)",
-					terminated ? "terminated" : "alive",
+					ProcessUtil.isProcessTerminated(port, processId) ? "terminated" : "alive",
 					ProcessUtil.isProcessAlive(processId),
 					ProcessUtil.isNetworkPortAvailable(port)));
 		}
 	}
+
 
 	/**
 	 * Standard termination method for platform process when time-out occurs.

--- a/graphalytics-core/src/main/java/science/atlarge/graphalytics/util/ProcessUtil.java
+++ b/graphalytics-core/src/main/java/science/atlarge/graphalytics/util/ProcessUtil.java
@@ -81,15 +81,21 @@ public class ProcessUtil {
     }
 
     public static void terminateProcess(Integer processId) throws Exception {
-        LOG.warn("Terminating process " + processId + " focibly.");
+        LOG.info("Terminating process " + processId + " forcibly.");
+
         Runtime runtime = Runtime.getRuntime();
         if (System.getProperty("os.name").toLowerCase().indexOf("windows") > -1) {
-            LOG.warn(String.format("Executing command \"taskkill %s\"", processId));
+            LOG.info(String.format("Executing command \"taskkill %s\"", processId));
             runtime.exec("taskkill " + processId);
         } else {
-            LOG.warn(String.format("Executing command \"kill -9 %s\"", processId));
+            LOG.info(String.format("Executing command \"kill -9 %s\"", processId));
             runtime.exec("kill -9 " + processId);
         }
+    }
+
+
+    public static boolean isProcessTerminated(int port, int processId) {
+        return isNetworkPortAvailable(port) && !isProcessAlive(processId);
     }
 
     public static void monitorProcess(Process process, String runId)  {

